apps/web/src/components/forms/2fa/authenticator-app-disable-dialog.tsx

```diff
@@ -0,0 +1,208 @@
+'use client';
+
+import { useState } from 'react';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import * as z from 'zod';
+
+import { ErrorCode, isErrorCode } from '@documenso/lib/next-auth/error-codes';
+import { trpc } from '@documenso/trpc/react';
+import { Button } from '@documenso/ui/primitives/button';
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@documenso/ui/primitives/dialog';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@documenso/ui/primitives/form/form';
+import { Input, PasswordInput } from '@documenso/ui/primitives/input';
+import { Textarea } from '@documenso/ui/primitives/textarea';
+import { useToast } from '@documenso/ui/primitives/use-toast';
+
+type AuthenticatorAppDisableDialogProps = {
+  open: boolean;
+  onOpenChange: (_val: boolean) => void;
+  onDisabled: () => void;
+};
+
+const formSchema = z.object({
+  password: z.string().min(6),
```

Comment:
Custom error messages for validation stuff would be really useful so users don't get the generic & meaningful ones. This comment applies to all Zod schemas.